packages/angular_devkit/schematics/src/tree/filtered.ts

```diff
@@ -38,3 +39,15 @@ export class FilteredTree extends VirtualTree {
     });
   }
 }
+
+export class FilterTree extends DelegateTree {
+  constructor(tree: Tree, filter: FilePredicate<boolean> = () => true) {
+    super(tree.branch());
+
+    tree.visit(path => {
+      if (!filter(path)) {
+        this.delete(path);
```

Comment:
This will delete files (as actions), no? A FilterTree should contains only actions that are pertained to files that match the pattern, unless you intend to do something else here. But if I have a tree that contains file1 and file2 and return FilterTree(x => x != file2), it would actually delete file2 which is probably not what we mean (and not what FilteredTree would have done).

I don’t see a test on actions of filtertree which would have caught this.

Reply:
New host tree specific filter added that does not use delete.